File: 2018/01.py
"""--- Day 1: Chronal Calibration ---"""

with open("input/01.txt") as f:
    freq = list(map(int, f.read().split("\n")[:-1]))
    c = set([0])
    a = i = 0
    while True:
        a += freq[i % len(freq)]
        if a in c:
            break
        c.add(a)
        i += 1
    print(sum(freq))
    # 536
    print(a)
    # 75108

File: 2019/01.js
// --- Day 1: The Tyranny of the Rocket Equation ---

const fs = require('fs')

function fuel (x) {
  return Math.max(0, Math.floor(x / 3) - 2)
}

function sum (x, y) {
  return x + y
}

fs.readFile('input/01.txt', { encoding: 'utf-8' }, (err, data) => {
  if (!err) {
    const lines = data
      .split('\n')
      .filter(x => x.length)
      .map(Number)

    const p1 = lines.map(fuel).reduce(sum)
    const p2 = lines.map(x => {
      let f = 0
      while ((x = fuel(x)) > 0) {
        f += x
      }
      return f
    }).reduce(sum)

    console.log(p1)
    console.log(p2)
  }
})
